gem/templatetags/gem_tags.py

```diff
@@ -14,6 +16,31 @@ def fieldtype(field):
     return field.field.widget.__class__.__name__
 
 
+@register.simple_tag(takes_context=True)
+def is_via_freebasics(context):
+    request = context['request']
+    return ('Internet.org' in request.META.get('HTTP_VIA', '') or
+            'InternetOrgApp' in request.META.get('HTTP_USER_AGENT', '') or
+            'true' in request.META.get('HTTP_X_IORG_FBS', ''))
+
+
+@register.inclusion_tag('core/tags/bannerpages.html', takes_context=True)
+def gembannerpages(context):
+    request = context['request']
+    locale = context.get('locale_code')
+
+    if request.site:
+        pages = request.site.root_page.specific.bannerpages().exact_type(
+            GemTextBanner)
+    else:
+        pages = []
+    return {
+        'bannerpages': get_pages(context, pages, locale),
+        'request': context['request'],
+        'locale_code': locale,
```

Comment:
Suggestion: if we are going to have a template tag for the banner you could just pass in the `is_via_freebasics: context['is_via_freebasics']` and not have the `is_via_freebasics` template tag. but happy to have the template tag if you prefer it that way